operator/pkg/ingress/envoy_config.go

```diff
@@ -392,8 +392,35 @@ func getRouteMatch(ingressPath slim_networkingv1.HTTPIngressPath) *envoy_config_
 	return nil
 }
 
+type SortableRoute []*envoy_config_route_v3.Route
+
+func (s SortableRoute) Len() int {
+	return len(s)
+}
+
+func (s SortableRoute) Less(i, j int) bool {
+	// Make sure Exact Match always comes first
+	isExactMatch1 := len(s[i].Match.GetPath()) != 0
+	isExactMatch2 := len(s[j].Match.GetPath()) != 0
+	if isExactMatch1 {
+		return true
+	}
+	if isExactMatch2 {
+		return false
+	}
+
+	// Make sure longest Prefix match always comes first
+	regexMatch1 := len(s[i].Match.GetSafeRegex().String())
+	regexMatch2 := len(s[j].Match.GetSafeRegex().String())
+	return regexMatch1 > regexMatch2
+}
+
+func (s SortableRoute) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+
 func getVirtualHost(ingress *slim_networkingv1.Ingress, rule slim_networkingv1.IngressRule) *envoy_config_route_v3.VirtualHost {
-	var routes []*envoy_config_route_v3.Route
+	var routes SortableRoute
```

Comment:
nit: if `rule.HTTP.Paths` length is known, you can initialize `routes` with `make(SortableRoute,0 , len(rule.HTTP.Paths))` to avoid superfluous memory allocations while appending on line 436

Reply:
good point :100: